File: Week_02/G20190343020121/n-ary-tree-postorder-traversal.java
class Solution {
    public List<Integer> postorder(Node root) {
      LinkedList<Node> stack  = new LinkedList<>();
        LinkedList<Integer> out = new LinkedList<>();
        if (root == null) {
             return out;
        }
        stack.add(root);
        while (!stack.isEmpty())  {
           Node  node = stack.pollLast();
           out.addFirst(node.val);
           for (Node item : node.children)  {
               if (item != null) {
                   stack.add(item);
               }
           }

        }
        return out;
    }
}

File: Week_02/G20190343020023/n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
    result = [];
    stack = [];
    otherStack = [];
    current = root;
    while (current) {
        if (current.children.length < 1 || otherStack.includes(current)) {
            result.push(current.val);
            current = stack.pop();
        }  else {
            let children = current.children;
            stack.push(current);
            otherStack.push(current);
            current = children[0];
            for (let i = children.length - 1; i > 0; i--) {
                stack.push(children[i]);
            }
        }
    }
    return result;
};